sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -207,6 +225,25 @@ case object PostgresDialect extends JdbcDialect {
       Some(StringType)
     } else if (sqlType == Types.OTHER && typeName.equals("jsonb")) {
       Some(StringType)
+    } else if (sqlType == Types.OTHER && typeName.equals("uuid")) {
+        Some(StringType)
+    } else if (sqlType == Types.ARRAY) {
+      typeName match {
```

Comment:
The underscores are particularly for the array types.  Postgres prepends them to all array types here https://github.com/pgjdbc/pgjdbc/blob/REL9_4_1204/org/postgresql/jdbc2/TypeInfoCache.java#L159